geowebcache/core/src/test/java/org/geowebcache/config/DefaultingConfigurationTest.java

```diff
@@ -0,0 +1,92 @@
+package org.geowebcache.config;
+
+import org.geowebcache.layer.TileLayer;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public abstract class DefaultingConfigurationTest extends ConfigurationTest<TileLayer, DefaultingConfiguration> {
```

Comment:
Due to the above, I don't think ConfigurationTest is actually usable for testing DefaultingConfiguration. You'll just need a unit test that verifies defaults get set correctly when setDefaultValues is called, plus an integration test of whatever uses that method (do a search for usages)